packages/lodestar/test/unit/network/gossip/validation/voluntaryExit.test.ts

```diff
@@ -0,0 +1,99 @@
+import {expect} from "chai";
+import sinon, {SinonStubbedInstance} from "sinon";
+
+import {config} from "@chainsafe/lodestar-config/lib/presets/minimal";
+import {EpochContext} from "@chainsafe/lodestar-beacon-state-transition";
+import * as validatorStatusUtils from "@chainsafe/lodestar-beacon-state-transition/lib/util/validatorStatus";
+import {ForkChoice} from "@chainsafe/lodestar-fork-choice";
+
+import {BeaconChain} from "../../../../../src/chain";
+import {StateRegenerator} from "../../../../../src/chain/regen";
+import {StubbedBeaconDb, StubbedChain} from "../../../../utils/stub";
+import {generateValidators} from "../../../../utils/validator";
+import {generateInitialMaxBalances} from "../../../../utils/balances";
+import {generateState} from "../../../../utils/state";
+import {generateEmptySignedVoluntaryExit} from "../../../../utils/attestation";
+import {validateGossipVoluntaryExit} from "../../../../../src/network/gossip/validation/voluntaryExit";
+import {VoluntaryExitErrorCode} from "../../../../../src/chain/errors/voluntaryExitError";
+
+describe("validate voluntary exit", () => {
+  const sandbox = sinon.createSandbox();
+  let dbStub: StubbedBeaconDb,
+    isValidIncomingVoluntaryExitStub: any,
+    chainStub: StubbedChain,
+    regenStub: SinonStubbedInstance<StateRegenerator>;
+
+  beforeEach(() => {
+    isValidIncomingVoluntaryExitStub = sandbox.stub(validatorStatusUtils, "isValidVoluntaryExit");
+    chainStub = (sandbox.createStubInstance(BeaconChain) as unknown) as StubbedChain;
```

Comment:
argh, StubbedChain is not well written but it's probably massive diff to rewrite to be as StubbedBeaconDb so ignore for now

Reply:
ok, i'll make an issue for this to help make sure it gets done in the future